### ISSUE TITLE
RemoteTech seems to break a lot of things - 1.1.2

### ISSUE THREAD

**Issue by apemanzilla** [KSP-KOS/KOS#2107]
With RemoteTech installed, there are a couple of strange issues I've found with kOS:

- I can't control the kOS terminal unless I have a probe core and a connection to KSC - having local control (a kerbal onboard) or a manned vessel with a connection to KSC doesn't work and just says `Signal lost. Waiting to re-acquire signal.`
- Steering and throttle control (`lock steering to ..., lock throttle to ...`) also doesn't seem to work unless there's a probe core and a connection to KSC. However, if there's a connection, steering and throttle are locked, and then the connection is lost, it continues to work.

**Comment by hvacengi** [comment]
I haven't used RT for a while now, but have used it extensively at one point.  I know that @Dunbaratu is using it as part of his realism overhaul game.

Because of how RT is structured, you must have a part with a signal processing unit installed on it in order to get the integration to work.  By default, probe cores are the only part that has this signal processing unit (I seem to recall the science lab might have it as well).  So, if you don't have a probe core you will never appear to have signal.  Not having a probe core will also prevent kOS from being able to interact with steering correctly, since it is unable to subscribe to the correct events.

That being said, I thought that we had things set up so that you could still interact with the terminal if you had a local control.  It is possible that logic is not working correctly.

Other than local interaction with the terminal, are all of your other issues solved by having a probe core installed (even without connection steering and throttle should work)?

**Comment by apemanzilla** [comment]
Everything else seems to work properly as long as there's a probe core attached. For now I whipped up [an MM config](https://gist.github.com/apemanzilla/c991e6ef205ce057f644fc4e085cc205) to add a signal processor to all kOS processors. There's still the local control bug, but that's not as important because most of what I use kOS for is done by boot files.

**Comment by apemanzilla** [comment]
Actually, sorry, with the SPU module added local control also works fine.

**Comment by hvacengi** [comment]
As @Dunbaratu and I spoke about this on slack, we're considering modifying the scope of this issue to include the request for optional MM patches to add SPU's to our kOS parts and/or an optional patch to add processors to probe cores.  Or something like that.  I've personally been reluctant to interact directly with modules from other mods, or to add our modules to other parts, because I don't want to introduce unexpected behaviors, especially those that happen as a result of an update.

The SPU requirement could at the very least be made more clear in the documentation.  This is a common issue raised, and it would be nice to provide more guidance to our users so they aren't as frustrated with something that is essentially working as designed.

[Note to self: go look for any other issues that are the same as this and link them all together]

**Comment by apemanzilla** [comment]
Yeah, adding `ModuleSPU` doesn't seem to be a great solution anyways - it causes RT to act a little strangely.

**Comment by hvacengi** [comment]
What kind of strange behaviors are you seeing?  If we add a documentation note about why we don't add SPU's I'd like to be able to reference what happens.

**Comment by apemanzilla** [comment]
- kOS control without a connection is finicky
- When setting a target there's sometimes a HUD message that the action couldn't be performed because there's no connection (even though there is a connection and the action does get performed properly)

**Comment by hvacengi** [comment]
And you've not had any of these issues if a probe core is installed with the regular SPU?  It seems odd that those issues might be connected to the part itself, though it could very well be that it assumes the part will be able to act as a control source.  Thanks for the info.

**Comment by apemanzilla** [comment]
The HUD messages go away, but disconnected control still doesn't always work. It's a pretty strange bug, and I've just taken to setting up satellite constellations to always be in contact.
